### PR TITLE
addon-links: add `LinkTo` component, and `hrefTo` function

### PR DESCRIPTION
**Pull request by Hypnosphi**
Issue: #1808

See the updated readme: https://github.com/storybooks/storybook/blob/39af55a076564ce1a2be6396f632bce30bdc4055/addons/links/README.md

### CODE REVIEW

File: examples/cra-kitchen-sink/src/stories/index.js [hunk 1]
```diff
@@ -40,7 +40,7 @@ const EVENTS = {
 const emiter = new EventEmiter();
 const emit = emiter.emit.bind(emiter);
 
-storiesOf('Welcome', module).add('to Storybook', () => <Welcome showApp={linkTo('Button')} />);
+storiesOf('Welcome', module).add('to Storybook', () => <Welcome showKind="Button" />);
```
Comment (Hypnosphi) [inline examples/cra-kitchen-sink/src/stories/index.js]: We can update it in CLI templates as soon as `3.3.0` gets released

File: lib/ui/src/modules/ui/components/routed_link.js [hunk 2]
```diff
@@ -1,46 +0,0 @@
-import PropTypes from 'prop-types';
```
Comment (Hypnosphi) [inline lib/ui/src/modules/ui/components/routed_link.js]: I forgot to remove these files when moving it to `components` in #1659

File: addons/links/README.md [hunk 3]
```diff
@@ -16,45 +16,100 @@ This addon works with Storybook for:
...
 
-First configure it as an addon by adding it to your addons.js file (located in the Storybook config directory).
+Then configure it as an addon by adding it to your addons.js file (located in the Storybook config directory).
 
 ```js
 import '@storybook/addon-links/register';
 ```
 
-Then you can import `linkTo` in your stories and use like this:
+## LinkTo component
```
Comment (ndelangen) [inline addons/links/README.md]: This is really react-specific.

Reading this and the issue this came from, adding this isn't really required right?
Can we do without it?

I'm mostly on-board with the `hrefTo` function. Although I kind of struggle to understand the use-case.

**Review by ndelangen — CHANGES_REQUESTED** [review]

File: addons/links/README.md (hunk 3 above)
Comment (pelotom) [reply addons/links/README.md]: > I'm mostly on-board with the `hrefTo` function. Although I kind of struggle to understand the use-case.

The use-case is when you have pure presentational components which take URLs as props and want them to link to other parts of your story.

Comment (Hypnosphi) [reply addons/links/README.md]: Yes it is react-specific, but it allows creating links which can open the story in new tab as well as in the same tab (without reloading the page). I can adjust `README` to indicate that this feature is react only. Users could implement similar components for other frameworks by combining `linkTo` and `hrefTo`

Comment (ndelangen) [reply addons/links/README.md]: Yes, `linkTo` already works cross-platform, I'm expecting `hrefTo` to work cross platform too.

Please update to readme to reflect this, and move the react-specific to the bottom of the 3, so the cross-platform stuff is at the top.

Comment (Hypnosphi) [reply addons/links/README.md]: I've made the adjustments

**Review by ndelangen — APPROVED** [review]

**Pull request merged**